src/controllers/MainController.js

```diff
@@ -0,0 +1,67 @@
+const Model = require('../Model/Model');
+const { 
+  readPlayerNum,
+  readRestartEnd,
+  printBallStrike,
+  printBall,
+  printStrike,
+  printNothing,
+  printThreeStrike,
+  exit,
+ } = require('../view/view');
+
+
+class MainController {
+  #model
+
+  mainGameController(){
+    this.#model = new Model()
+    this.sendPlayerNum();
+  }
+
+  sendPlayerNum(){
+    readPlayerNum(this.ballCountController.bind(this));
+  }
+
+  ballCountController(input){
+    this.#model.savePlayerNum(input);
+    if (this.#model.getOpponentNum() !== this.#model.getPlayerNum()){
+      this.ballCountOutputControll()
+      this.sendPlayerNum();
+    }
+    if (this.#model.getOpponentNum() === this.#model.getPlayerNum()) {
+      printThreeStrike();
+      this.endController();
+    } 
+  }
+
+  ballCountOutputControll(){
+    if (this.#model.getBall() > 0 && this.#model.getStrike() > 0) {
+      printBallStrike(this.#model.getBall(), this.#model.getStrike()); 
+    }
+    if (this.#model.getBall() > 0 && this.#model.getStrike() === 0) {
+      printBall(this.#model.getBall());
+    }
+    if (this.#model.getBall() === 0 && this.#model.getStrike() > 0) {
+      printStrike(this.#model.getStrike());
+    }
+    if (this.#model.getBall() === 0 && this.#model.getStrike() ===0) {
+      printNothing();
+    }
+  }
```

Comment:
어차피 this.#model.getBall(), this.#model.getStrike() 를 계속 사용할거라면 그냥 변수로 빼고 그거 쓰세요

```suggestion
  ballCountOutputControll(){
    const ballCount = this.#model.getBall();
    const strikeCount = this.#model.getStrike();
    
    if (ballCount > 0 && strikeCount > 0) {
      printBallStrike(ballCount, strikeCount); 
    }
    if (ballCount > 0 && strikeCount === 0) {
      printBall(ballCount);
    }
    if (ballCount === 0 && strikeCount > 0) {
      printStrike(strikeCount);
    }
    if (ballCount === 0 && strikeCount ===0) {
      printNothing();
    }
  }
```